kyaml/fn/framework/patch.go

```diff
@@ -168,8 +168,7 @@ func (cpt ContainerPatchTemplate) apply(matches []*yaml.RNode) error {
 	}
 
 	for i := range matches {
-		// TODO(knverey): Make this work for more Kinds and expose the helper for doing so.
-		containers, err := matches[i].Pipe(yaml.Lookup("spec", "template", "spec", "containers"))
+		containers, err := yaml.LookupFirstMatch(yaml.ConventionalContainerPaths).Filter(matches[i])
```

Comment:
What is the motivation for changing the way the filter is invoked? I believe you could have still used `matches[i[.Pipe(yaml.LookupFirstMatch...` so I'm wondering if there is a reason you prefer to call `Filter` directly rather than using `Pipe`?

Reply:
LGTM other than this, but I'd like to understand the reasoning.

Reply:
No particular reason. As you noted, my helper originally wasn't a filter, so when I made it one, I fixed this call site by tacking `Filter` onto the end without looking back at the original diff. The only difference if you only have one Filter seems to be that `Pipe` short-circuits when given `nil` input, and special-cases root document nodes. Doesn't hurt to use it here, or in the helper. Since it raised questions for you, I'll change it.